modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/BPlusTree.java

```diff
@@ -1826,6 +1827,23 @@ public final boolean removex(L row) throws IgniteCheckedException {
         return res != null ? res : false;
     }
 
+    /**
+     * Returns a cursor that removes rows when being iterated.
+     * Uses a specified closure as a predicate to filter rows.
+     *
+     * @param clo TreeRowClosure Closure to filter rows.
+     * @throws IgniteCheckedException If failed.
+     */
+    public GridCursor<Void> purge(TreeRowClosure<L, T> clo) throws IgniteCheckedException {
```

Comment:
I don't get why we should share "cursor that removes rows when being iterated"?
Can we implement regular `Iterator` with support of `remove` operation?